src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MarkJavaObjects.cs

```diff
@@ -276,12 +276,20 @@ static bool IsUserType (TypeDefinition type)
 
 		void PreserveImplementor (TypeDefinition type)
 		{
+#if ILLINK
+			if (!type.HasInterfaces)
+				return;
+			foreach (var iface in type.Interfaces) {
+				Annotations.Mark (iface.InterfaceType.Resolve ());
+			}
+#else  // !ILLINK
```

Comment:
@jonpryor I talked a bit with @sbomer, and basically what was happening is that in .NET 6+ only types that are already marked come into this step. In Xamarin.Android, all types come in -- which was the cause of the size regression. Additionally, only .NET 6+ will remove interfaces from types.

Do you think it's worth me deleting the `#else` block here? or just leave it?

https://github.com/xamarin/xamarin-android/blob/4945f50c8d26759151647f489757f12ea5cf29f3/src/Xamarin.Android.Build.Tasks/Linker/MonoDroid.Tuner/MarkJavaObjects.cs#L285-L292

I think we thought this was completely "dead" code that didn't do anything.

Reply:
We need to figure out how to prevent .NET 6 from removing Java interfaces for `Java.Lang.Object` & `Java.Lang.Throwable` subclasses.

Regarding the `#else` block in `MarkJavaObject.cs`, it's still *weird*/wrong; it shouldn't preserve *only* methods ending in `Handler`, it should preserve everything (shouldn't it?).  It should also preserve the implemented interfaces (though apparently those are already implicitly preserved?).

Consider `src/Mono.Android/obj/Debug/monoandroid10/android-33/mcw/Android.App.ActionBar.cs`:

```csharp
		[global::Android.Runtime.Register ("mono/android/app/ActionBar_OnMenuVisibilityListenerImplementor")]
		internal sealed partial class IOnMenuVisibilityListenerImplementor : global::Java.Lang.Object, IOnMenuVisibilityListener {

			object sender;

			public IOnMenuVisibilityListenerImplementor (object sender) : base (global::Android.Runtime.JNIEnv.StartCreateInstance ("mono/android/app/ActionBar_OnMenuVisibilityListenerImplementor", "()V"), JniHandleOwnership.TransferLocalRef)
			{
				global::Android.Runtime.JNIEnv.FinishCreateInstance (((global::Java.Lang.Object) this).Handle, "()V");
				this.sender = sender;
			}

			#pragma warning disable 0649
			public EventHandler<MenuVisibilityEventArgs>? Handler;
			#pragma warning restore 0649

			public void OnMenuVisibilityChanged (bool isVisible)
			{
				var __h = Handler;
				if (__h != null)
					__h (sender, new MenuVisibilityEventArgs (isVisible));
			}

			internal static bool __IsEmpty (IOnMenuVisibilityListenerImplementor value)
			{
				return value.Handler == null;
			}
		}
```

*If* `IOnMenuVisibilityListenerImplementor` is referenced, then *everything within it* should be preserved, not just the `add_Handler()` and `remove_Handler()` methods.